pkg/core/blockchain.go

```diff
@@ -1199,9 +1196,25 @@ func (bc *Blockchain) verifyTx(t *transaction.Transaction, block *block.Block) e
 	if t.ValidUntilBlock <= height || t.ValidUntilBlock > height+transaction.MaxValidUntilBlockIncrement {
 		return errors.Errorf("transaction has expired. ValidUntilBlock = %d, current height = %d", t.ValidUntilBlock, height)
 	}
+	balance := util.Fixed8FromInt64(bc.GetNEP5Balances(t.Sender).Trackers[bc.contracts.GAS.Hash].Balance)
+	totalSenderFee := util.Fixed8FromInt64(0)
+	for _, txWithFee := range bc.memPool.GetVerifiedTransactions() {
+		if txWithFee.Tx.Sender.Equals(t.Sender) {
+			totalSenderFee += txWithFee.Tx.NetworkFee + txWithFee.Tx.SystemFee
+		}
+	}
```

Comment:
This part should be handled in mempool directly, at the moment this check is racy. In `verifyTx` it only makes sense doing some trivial `balance <  t.SystemFee + t.NetworkFee` check (it doesn't cost a lot).

Reply:
Think of this check as of double input spend check for UTXO transactions.

Reply:
I don't see this check in the mempool, am I missing something?

Reply:
Should we put this check into `(*Pool) Verify(tx)` method?

Reply:
`checkTxConflicts`

Reply:
Solved this by adding `sendersFeeMonitor` to mempool